libcudacxx/include/cuda/ptx

```diff
@@ -0,0 +1,352 @@
+
+// cp.reduce.async.bulk
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-reduce-async-bulk
+
+// cp.async.bulk.tensor
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+
+// cp.reduce.async.bulk.tensor
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-reduce-async-bulk-tensor
+
+// cp.async.bulk.commit_group
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-commit-group
+
+// cp.async.bulk.wait_group
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-wait-group
+
+
+// Lower priority:
+
+// prefetch{.tensormap_space}.tensormap [a];  // prefetch the tensormap
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-prefetch-prefetchu
+
+// cp.async.bulk.prefetch
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk
+
+// cp.async.bulk.prefetch.tensor
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-prefetch-tensor
+
+/*
+ *  Shared memory barrier
+ *
+ */
+
+// mbarrier.expect_tx
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-expect-tx
+
+// mbarrier.complete_tx
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-complete-tx
+
+// mbarrier.arrive.expect_tx
+// Support for count argument without the modifier .noComplete requires sm_90 or higher.
+// Qualifier .expect_tx requires sm_90 or higher.
+// Sub-qualifier ::cluster requires sm_90 or higher.
+// Support for .cluster scope requires sm_90 or higher.
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-arrive
+
+
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-arrive
+#if (defined(__CUDA_MINIMUM_ARCH__) && 900 <= __CUDA_MINIMUM_ARCH__) || (!defined(__CUDA_MINIMUM_ARCH__))
+template <dot_scope _Sco>
+_LIBCUDACXX_DEVICE inline
+_CUDA_VSTD::uint64_t mbarrier_arrive_expect_tx(sem_release_t __sem, scope_t<_Sco> __scope, space_shared_t __spc, _CUDA_VSTD::uint64_t* __addr, _CUDA_VSTD::uint32_t __tx_count)
+{
+    // Arrive on local shared memory barrier
+    static_assert(__scope == scope_cta || __scope == scope_cluster, "");
+    _CUDA_VSTD::uint64_t __token;
+
+    if constexpr (__scope == scope_cta) {
```

Comment:
This is a C++17 only feature

Reply:
I changed this to `if _LIBCUDACXX_CONSTEXPR_AFTER_CXX14`. That is the correct macro for C++17-style `constexpr` right?